internal/pkg/cli/app_init.go

```diff
@@ -299,15 +314,46 @@ func (o *initAppOpts) askDockerfile() error {
 }
 
 func (o *initAppOpts) askAppPort() error {
+	// Use flag before anything else
 	if o.AppPort != 0 {
 		return nil
 	}
 
+	log.Infof(fmtParsePortFromDockerfileStart,
+		color.HighlightUserInput(o.DockerfilePath),
+		color.HighlightUserInput(o.AppName),
+	)
+
+	o.setupParser(o)
+	ports, err := o.df.GetExposedPorts()
+	// Ignore any errors in dockerfile parsing--we'll use the default instead.
+	if err != nil {
+		log.Debugln(err.Error())
+	}
+	var defaultPort = defaultAppPortString
+	switch len(ports) {
+	case 0:
+		log.Infof(fmtParseFromDockerfileNoPort,
+			color.HighlightUserInput(o.AppName),
+		)
+	case 1:
+		o.AppPort = ports[0]
+		log.Successf(fmtParsePortFromDockerfileComplete,
+			color.HighlightUserInput(strconv.Itoa(int(o.AppPort))),
+		)
+	default:
+		defaultPort = strconv.Itoa(int(ports[0]))
+		log.Infoln(parseFromDockerfileTooManyPorts)
+	}
+
+	if o.AppPort != 0 {
+		return nil
+	}
```

Comment:
Do you think we can get rid of this if-statement if we return `nil` for `case 1`?

Reply:
yes, definitely. Resolved in latest commit.